src/common/CExpression.cpp

```diff
@@ -249,13 +249,23 @@ bool IsValidGameObjDef( lpctstr pszTest )
 {
 	if (!IsSimpleNumberString(pszTest))
 	{
+		/*
+		 * We need to check if defname is exist first, and get the value to override pszTest lpctstr to let people can add item from defname.
+		 * Check issue #357 for more information (https://github.com/Sphereserver/Source-X/issues/357)
+		 * xwerswoodx
+		 */
+		const CVarDefCont * pDefBase = g_Exp.m_VarDefs.GetKey(pszTest);

```

Comment:
I removed that when making RESDEF.
Before that, both DEFNAME section data and item/char defnames were stored in g_Exp.m_VarDefs. That was inefficient, because if you are searching for a DEFNAME, you'll have to search for it between thousands of item/char defnames.
I didn't think about backwards compatibility defs though, so i guess i'll have to add soon a RESDEF section, similar to the DEFNAME one.

Reply:
Hmm, I see "m_VarResDefs" already exist for Resource Defs, but it doesn't include Defnames inside it, but adding them into m_VarResDefs can cause more problems i think, maybe as you say creating RESDEF is better solution for it. I thought it can be problem to check all vars all the time but "IsValidGameObjDef" boolean only called under .add command so I don't know if it should change but RESDEF could be better for items because item IDs and MORE1 etc also check defnames and every item does thousands of check all the time. But this boolean only called under .add command so this can be temporary solution for .add, if you want I can remove it, but I think this can stay until RESDEF added, then we can change it to RESDEF instead of checking vars. But checking vars for .add, which called by client and not used all the time, won't cause much issues i think.

Reply:
I am waiting your answer to change that, i leave that until resdef added, then we can remove it. It's just for ADD command so i think it doesn't cause much issue, but if you want, i can cancel this and pull again :) Other requests done.

Reply:
The problem may be happening only with .add (not sure) because there are other workarounds (checking both DEFs and RESDEFs) in some other sections of the script parsing code, but as said those are workarounds, ideally removed in future, and we shouldn't rely totally on them. I can do some testing and then send the new RESDEFNAME section commit!

Reply:
This is already removed with the latest commit so I think I can resolve this conversation.